x-pack/legacy/plugins/ml/server/routes/job_service.ts

```diff
@@ -0,0 +1,611 @@
+      try {
+        const { jobsSummary } = jobServiceProvider(context.ml!.mlClient.callAsCurrentUser);
+        const { jobIds } = request.body;
+        const resp = await jobsSummary(jobIds);
+
+        return response.ok({
+          body: resp,
+        });
+      } catch (e) {
+        return response.customError(wrapError(e));
+      }
+    })
+  );
+
+  /**
+   * @apiGroup JobService
+   *
+   * @api {post} /api/ml/jobs/jobs_with_timerange
+   * @apiName JobsWithTimerange
+   * @apiDescription Creates a list of jobs with data about the job's timerange
+   */
+  router.post(
+    {
+      path: '/api/ml/jobs/jobs_with_timerange',
+      validate: {
+        body: schema.object(jobsWithTimerangeSchema),
+      },
+    },
+    licensePreRoutingFactory(xpackMainPlugin, async (context, request, response) => {
+      try {
+        const { jobsWithTimerange } = jobServiceProvider(context.ml!.mlClient.callAsCurrentUser);
+        const { dateFormatTz } = request.body;
+        const resp = await jobsWithTimerange(dateFormatTz);
+
+        return response.ok({
+          body: resp,
+        });
+      } catch (e) {
+        return response.customError(wrapError(e));
+      }
+    })
+  );
+
+  /**
+   * @apiGroup JobService
+   *
+   * @api {post} /api/ml/jobs/jobs
+   * @apiName CreateFullJobsList
+   * @apiDescription Creates a list of jobs
+   */
+  router.post(
+    {
+      path: '/api/ml/jobs/jobs',
+      validate: {
+        body: schema.object(jobIdsSchema),
+      },
+    },
+    licensePreRoutingFactory(xpackMainPlugin, async (context, request, response) => {
+      try {
+        const { createFullJobsList } = jobServiceProvider(context.ml!.mlClient.callAsCurrentUser);
+        const { jobIds } = request.body;
+        const resp = await createFullJobsList(jobIds);
+
+        return response.ok({
+          body: resp,
+        });
+      } catch (e) {
+        return response.customError(wrapError(e));
+      }
+    })
+  );
+
+  /**
+   * @apiGroup JobService
+   *
+   * @api {get} /api/ml/jobs/groups
+   * @apiName GetAllGroups
+   * @apiDescription Returns array of group objects with job ids listed for each group
+   */
+  router.get(
+    {
+      path: '/api/ml/jobs/groups',
+      validate: false,
+    },
+    licensePreRoutingFactory(xpackMainPlugin, async (context, request, response) => {
+      try {
+        const { getAllGroups } = jobServiceProvider(context.ml!.mlClient.callAsCurrentUser);
+        const resp = await getAllGroups();
+
+        return response.ok({
+          body: resp,
+        });
+      } catch (e) {
+        return response.customError(wrapError(e));
+      }
+    })
+  );
+
+  /**
+   * @apiGroup JobService
+   *
+   * @api {post} /api/ml/jobs/update_groups
+   * @apiName UpdateGroups
+   * @apiDescription Updates 'groups' property of an anomaly detection job
+   */
+  router.post(
+    {
+      path: '/api/ml/jobs/update_groups',
```

Comment:
I get an error from this endpoint when trying to add two groups to two jobs in the Jobs list:

![image](https://user-images.githubusercontent.com/7405507/74354976-14677d00-4db4-11ea-893b-cc9110be7584.png)

Reply:
Updated schema to fix this in 9d1014383ec15daef0ec6402379d54d412360d15